### PR TITLE
feat(virtual-scroll): remove virtual scroll component

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

`ion-virtual-scroll` is deprecated and needs to be removed with v7.

Issue URL: Internal

## What is the new behavior?

Removes all references to `ion-virtual-scroll` from `core`, `angular`, `react`, `vue` and `angular-server`. 

- Removes `ion-virtual-scroll` component
- Removes internal virtual scroll components available on the `IonicModule` (Angular only)

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

Developers will need to migrate all usages of `ion-virtual-scroll` to a supported framework integration:
- [Angular](https://ionicframework.com/docs/angular/virtual-scroll)
- [React](https://ionicframework.com/docs/react/virtual-scroll)
- [Vue](https://ionicframework.com/docs/vue/virtual-scroll)

Developers will need to remove/update any references to virtual scroll types: `Cell`, `VirtualNode`, `CellType`, `NodeChange`, `HeaderFn`, `ItemHeightFn`, `HeaderHeightFn`, `FooterHeightFn`, `ItemRenderFn`, and `DomRenderFn` that are imported from `@ionic/core`. 

## Other information

Dev-build: `6.2.4-dev.11661284341.17fdd471`

### CODE REVIEW

**Review by liamdebeasi — APPROVED** [review]
The code changes look good. Do you have a PR for the documentation changes?

**Comment by sean-perkins** [comment]
@liamdebeasi working on that next 👍 I will merge once both are available/approved.

**Comment by sean-perkins** [comment]
Docs PR is available here: https://github.com/ionic-team/ionic-docs/pull/2507

**Review by amandaejohnston — APPROVED** [review]

**Pull request merged**
